### ISSUE TITLE
Travis environment lacks llvm-8

### ISSUE THREAD

**Issue by kimgr** [include-what-you-use/include-what-you-use#642]
Recent Travis build have all failed with
```
The following packages have unmet dependencies:
 clang : Depends: clang-8 (>= 8~) but it is not installable
 libclang-dev : Depends: libclang-8-dev (>= 8~) but it is not installable
 llvm-dev : Depends: llvm-runtime (= 1:8.0-45+0~20181005115440.34~1.gbp3de5db) but 1:3.4-0ubuntu1 is to be installed
            Depends: llvm-8-dev (>= 8~) but it is not installable
E: Unable to correct problems, you have held broken packages.
```
see e.g. https://travis-ci.org/include-what-you-use/include-what-you-use/builds/481709235?utm_medium=notification&utm_source=email

I'm not sure if this is caused by the recent LLVM 8 release branch, or Trusty now being unsupported, or something else. I don't have enough Travis chops to solve this in short order, so I thought I'd open an issue to track it.

**Comment by jru** [comment]
Hey,

It really seems the problem is on their end.
I queried the repository `deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty main` and all specific packages for version 8 are missing.

For example, this is the definition for the `clang` package:

```
Package: clang
Source: llvm-defaults (0.48+0~20190119150002.36~1.gbp9f1486)
Version: 1:8.0-48+0~20190119150002.36~1.gbp9f1486
Architecture: amd64
Maintainer: LLVM Packaging Team <pkg-llvm-team@lists.alioth.debian.org>
Installed-Size: 43
Depends: clang-8 (>= 8~)
Breaks: clang-3.2, clang-3.3, clang-3.4 (<< 1:3.4.2-7~exp1), clang-3.5 (<< 1:3.5~+rc1-3~exp1)
Replaces: clang (<< 3.2-1~exp2), clang-3.2, clang-3.3, clang-3.4 (<< 1:3.4.2-7~exp1), clang-3.5 (<< 1:3.5~+rc1-3~exp1)
Priority: optional
Section: devel
Filename: pool/main/l/llvm-defaults/clang_8.0-48+0~20190119150002.36~1.gbp9f1486_amd64.deb
Size: 7602
SHA256: 409dd1dafec3d25193deefb0e14e32ee4850124799f589c6632d0043d9283bfe
SHA1: 745f0f8dc1a3704e60a892069e6f6bd06de6d11f
MD5sum: 21b46dcd8161a6b65b7b555f14dc1202
Description: C, C++ and Objective-C compiler (LLVM based)
 Clang project is a C, C++, Objective C and Objective C++ front-end
 for the LLVM compiler. Its goal is to offer a replacement to the GNU Compiler
 Collection (GCC).
 .
 Clang implements all of the ISO C++ 1998, 11 and 14 standards and also
 provides most of the support of C++17.
 .
 This is a dependency package providing the default clang compiler.
```

As you can see, it mentions `clang-8 (>= 8~)` as a dependency, but there is no such package in the repository.

I guess, we can either report it or just wait until it gets looked into.

**Comment by kimgr** [comment]
Thanks!

> I queried the repository `deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty main` and all specific
> packages for version 8 are missing.

What command did you use to do that? Assuming `deb` is not a command -- if it is, which package could provide it for my Ubuntu system? I've never fully gotten my head around Linux packaging systems, it would be nice to learn more.

Browsing around the raw structure at http://apt.llvm.org/trusty/, there is a https://apt.llvm.org/trusty/pool/main/l/llvm-toolchain-8/ directory, but this seems to be for the 8.0.0 release, and a https://apt.llvm.org/trusty/pool/main/l/llvm-toolchain-snapshot/ directory for trunk.

I think we really want the latter for IWYU master, but I'm not sure if there's a way to spell the package name to force that.

**Comment by jru** [comment]
I actually inspected the `Packages` file from: 
http://apt.llvm.org/trusty/dists/llvm-toolchain-trusty/main/binary-amd64/Packages
But there is also, `apt-cache show clang` that, I believe, does a similar thing.

And I also tried it on a trusty docker image with the same results as the Travis build:
```sh
$ docker run --rm -it ubuntu:trusty bash
$ echo "deb http://apt.llvm.org/trusty/ llvm-toolchain-trusty main" >> /etc/apt/sources.list
$ apt update
$ apt install clang
```

The Travis history shows that the [last build](https://travis-ci.org/include-what-you-use/include-what-you-use/builds/480606521#L470) that managed to fetch the *8.0* packages was from 4 days ago, before the release 9.0 started, I think. 

If that's correct, then after the release 8.0 was cut, we should start getting 9.0 packages automatically, but somehow we don't..

The news from apt.llvm.org also suggest that somehing was recently changed in the packages:
```
Jan 19th 2019 - Branch 8 created
Jan 19th 2019 - Ubuntu Disco (19.10) support
Jan 19th 2019 - Artful jobs disabled (but packages still available)
```

> I think we really want the latter for IWYU master, but I'm not sure if there's a way to spell the package name to force that.

Not really sure, either. The thing is that before, the default packages seemed to always point to the snapshot version (some days/weeks old) without needing to specify any version at all.
